imagetest/utils/test_utils.go

```diff
@@ -201,7 +125,9 @@ func DownloadPrivateKey(user string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	sourcesPath, err := GetMetadataAttribute("daisy-sources-path")
+
+	// TODO: pass down the main context here
+	sourcesPath, err := GetMetadata(context.Background(), "instance", "attributes", "daisy-sources-path")
```

Comment:
I think it's better to have the `GetMetadata` method itself call `context.Background()` instead of having to pass it in. It'll make the method more intuitive to call.

Reply:
The idea is to never have anyone calling `GetMetadata()` with a `context.Background()` instead using the main process's context, hence the comment right above (on line 129). Having `GetMetadata()` creating its own context will only lead to disasters due to no context cancelation strategy.

Reply:
I introduced a deadline based context cancelation and removed the TODOS and this context creation inside these util functions. Thanks.